src/LiveComponent/src/EventListener/QueryStringInitializeSubscriber.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\LiveComponent\EventListener;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\UX\LiveComponent\Metadata\LiveComponentMetadata;
+use Symfony\UX\LiveComponent\Metadata\LiveComponentMetadataFactory;
+use Symfony\UX\LiveComponent\Util\QueryStringPropsExtractor;
+use Symfony\UX\TwigComponent\Event\PreMountEvent;
+
+/**
+ * @author Nicolas Rigaud <squrious@protonmail.com>
+ *
+ * @experimental
+ *
+ * @internal
+ */
+class QueryStringInitializeSubscriber implements EventSubscriberInterface
+{
+    public function __construct(
+        private readonly RequestStack $requestStack,
+        private readonly LiveComponentMetadataFactory $metadataFactory,
+        private readonly QueryStringPropsExtractor $queryStringPropsExtractor,
+    ) {
+    }
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            PreMountEvent::class => 'onPreMount',
+        ];
+    }
+
+    public function onPreMount(PreMountEvent $event): void
+    {
+        $request = $this->requestStack->getMainRequest();
+
+        if (null === $request) {
+            return;
+        }
+
+        $metadata = new LiveComponentMetadata(
+            $event->getMetadata(),
+            $this->metadataFactory->createPropMetadatas(new \ReflectionClass($event->getComponent()::class))
+        );
+
+        if (!$metadata->hasQueryStringBindings()) {
+            return;
+        }
+
+        $queryStringData = $this->queryStringPropsExtractor->extract($request, $metadata, $event->getComponent());
+
+        $event->setData(array_merge($event->getData(), $queryStringData));
```

Comment:
I'm not sure this should be done there. Or at leasst without additional safeguards.

Maybe i missread the code, but if the live component uses ComponentValidator for example,.... it will not be called right now ?

Because the data set there is the data which will be used to render the component, to write in HTML, etc..

I may be wrong... but i'm pretty sure there is a big security risk there.

Reply:
Anyone can set a value by writing something in the URL... that seems enough to validate, above all if the user added some Assert on that prop

But maybe it's on the ComponentValidator there is something to do indeed

Reply:
Consider the following:

```php
// ...

class MyController extends AbstractController {
    
    #[Route('/{title}')]
    public function index(Environment $twig, string $title): Response
    {
        $template = '<twig:MyComponent title="{{ title }}"></twig:MyComponent>';

        $view = $twig->createTemplate($template)->render(['title' => $title]);
        
        return new Response($view);
    } 
}
```

This is, imo, something valid. If `MyComponent` is a regular TwigComponent, it can not even use the validator from LiveComponent. If the developers want to validate data, they have to implement their own logic in a PostMount hook.

I'm not found of setting up automatic validation in the specific case of a LiveComponent whom props have been set from the query string:
- we'll need to track these props from the `PreMountEvent`
- on `PostMountEvent`, we'll have to get these props back, ensure the current values are the ones set on PreMount, use reflection to determine if the component has validation mechanisms, then trigger validation

I'd rather add a documentation entry in this feature doc, suggesting the use of validation in a PostMount hook.

Reply:
The developper here made an explicit error himself there. We indeed cannot do anything.

But the second you propose a single on/off attribute in a class, people will use it without thinking there may be consequences. 

For me there is a real danger here (similar to what happened with Autocomplete). We'll see and i hope you are right and that will never happen.